compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTypeGen.java

```diff
@@ -1856,13 +1932,28 @@ private void loadTupleType(MethodVisitor mv, BTupleType bType) {
      * @param bType user defined type
      */
     private void loadUserDefinedType(MethodVisitor mv, BType bType) {
+        BTypeSymbol typeSymbol = bType.tsymbol.isTypeParamResolved ? bType.tsymbol.typeParamTSymbol : bType.tsymbol;
+        BType typeToLoad = bType.tsymbol.isTypeParamResolved ? typeSymbol.type : bType;
```

Comment:
I think this and above line can be merged `BType typeToLoad = bType.tsymbol.isTypeParamResolved ? bType.tsymbol.typeParamTSymbol.type : bType;`